ironfish-rust/src/merkle_note.rs

```diff
@@ -301,11 +299,7 @@ mod test {
         );
         let diffie_hellman_keys = EphemeralKeyPair::new();
 
-        let value_commitment = ValueCommitment {
-            value: note.value,
-            randomness: jubjub::Fr::random(thread_rng()),
-            asset_generator: note.asset_generator(),
-        };
+        let value_commitment = ValueCommitment::new(note.value, note.asset_generator());
```

Comment:
I'm not totally opposed to hiding this randomness but I think we should be careful putting it in the background. It could be helpful to explicitly show where we are getting randomness from or that a certain component depends on randomness.

Reply:
Hmm, I'd have to think about this more to form a better opinion, but my initial reaction is that randomness is a pretty minor implementation detail that doesn't need to be thought about unless you're working on the proofs or something.